### DISCUSSION TITLE
Compiling RP2040/CMakeLists.txt fails on changing SECOND_SEGMENT_INDEX to ON

### DISCUSSION THREAD

**Discussion by delavandaag** [awawa-dev/HyperSPI#57] [Q&A] [answered]
I am trying to change and recompile RP2040/CMakeLists.txt after having created a fork. For my Adafruit Feather SCORPIO RP2040 I have to change OUTPUT_DATA_PIN to 16, SECOND_SEGMENT_INDEX to ON and SECOND_SEGMENT_REVERSED to ON. As I started, every attempt to recompile the amended file failed. I then started with changing only a comment, which succeeded. Also the subsequent change of BOOT_WORKAROUND to OFF was successful, and the third change of OUTPUT_DATA_PIN to 16 was successful as well. Upon trying to also change the SECOND_SEGMENT_INDEX to ON, it failed. I got the following error (lines 113 - 145 of 'Build packages'):

`-- Build files have been written to: /home/runner/work/HyperSPI/HyperSPI/rp2040/build/elf2uf2
[  6%] Performing build step for 'ELF2UF2Build'
[ 50%] Building CXX object CMakeFiles/elf2uf2.dir/main.cpp.o
[100%] Linking CXX executable elf2uf2
[100%] Built target elf2uf2
[  6%] No install step for 'ELF2UF2Build'
[  6%] Completed 'ELF2UF2Build'
[  6%] Built target ELF2UF2Build
[  6%] Generating /home/runner/work/HyperSPI/HyperSPI/rp2040/HyperSerialPico/generated/neopixel.pio.h
[  6%] Built target HyperSPI_Pico_sk6812Cold_multisegment_at_ON_neopixel_pio_h
[  7%] Building CXX object HyperSerialPico/CMakeFiles/HyperSPI_Pico_sk6812Cold_multisegment_at_ON.dir/__/hyperspi.cpp.obj
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:65:54: note: '#pragma message: NEOPIXEL_RGBW = 1'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:71:51: note: '#pragma message: COLD_WHITE = 1'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:80:41: note: '#pragma message: DATA_PIN = 16'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:81:46: note: '#pragma message: SPI_INTERFACE = ((spi_inst_t *)((spi_hw_t *)0x4003c000u))'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:82:45: note: '#pragma message: SPI_DATA_PIN = 4'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:83:46: note: '#pragma message: SPI_CLOCK_PIN = 2'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:84:48: note: '#pragma message: SPI_CHIP_SELECT = 5'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:87:59: note: '#pragma message: Using parallel mode for segments'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:101:51: note: '#pragma message: LED_DRIVER = sk6812p'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:102:67: note: '#pragma message: SECOND_SEGMENT_START_INDEX = ON'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:103:52: note: '#pragma message: LED_DRIVER2 = sk6812p'
/home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:104:64: note: '#pragma message: SECOND_SEGMENT_REVERSED = SECOND_SEGMENT_REVERSED'
In file included from /home/runner/work/HyperSPI/HyperSPI/rp2040/HyperSerialPico/include/main.h:36,
                 from /home/runner/work/HyperSPI/HyperSPI/rp2040/hyperspi.cpp:[116](https://github.com/delavandaag/HyperSPI/actions/runs/10796408542/job/29945337265#step:4:117):
/home/runner/work/HyperSPI/HyperSPI/rp2040/HyperSerialPico/include/base.h: In member function 'void Base::initLedStrip(int)':
/home/runner/work/HyperSPI/HyperSPI/rp2040/HyperSerialPico/include/base.h:88:50: error: 'ON' was not declared in this scope
/home/runner/work/HyperSPI/HyperSPI/rp2040/HyperSerialPico/include/base.h: In member function 'bool Base::setStripPixel(uint16_t, ColorDefinition&)':
/home/runner/work/HyperSPI/HyperSPI/rp2040/HyperSerialPico/include/base.h:153:51: error: 'ON' was not declared in this scope
gmake[2]: *** [HyperSerialPico/CMakeFiles/HyperSPI_Pico_sk6812Cold_multisegment_at_ON.dir/build.make:76: HyperSerialPico/CMakeFiles/HyperSPI_Pico_sk6812Cold_multisegment_at_ON.dir/__/hyperspi.cpp.obj] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1754: HyperSerialPico/CMakeFiles/HyperSPI_Pico_sk6812Cold_multisegment_at_ON.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
Error: Process completed with exit code 2.`

I have not yet tried to change the SECOND_SEGMENT_REVERSED to ON.

Is this a bug or am I doing something wrong? Somebody who can reproduce this?

**Comment by awawa-dev** [comment, marked as answer]
```
# Use multi-segment, starting index of second led strip or OFF to disable
set(SECOND_SEGMENT_INDEX ON)
```

it must be a number (starting index of the second segment), not `ON`

**Reply by delavandaag** [reply]
Ohw ... Yeah, I could have thought of that myself. Thanks again, Awawa!
